contentful/template-blog-webapp-nextjs
# feat: [PLATO-409] Segment.io integration

**_What will change?_**

**Segment/typewriter tracking**
Added Segment / Typewriter for analytics purposes

More intricate documentation is available through the readme file that was added in the analytics folder, but in a nutshell:
- Added a boilerplate for generating a typewriter.yml file, which is then used to generate a typed typewriter.analytics object.
- Sentry is set uo to report any Segment schema violation
- The configuration of Segment is driven by environment variables
- The typewriter schema is automatically regenerated on various events

**Consent management**
- Added @contentful/experience-consent-manager for Contentful's Osano based consent management
- Built a custom RQ hook that creates a ConsentManager instance and exposes it globally
- Added a little component that injects the custom styling, and adds a button to open the consent settings drawer

Ticket: https://contentful.atlassian.net/browse/PLATO-409
Docs: https://contentful.atlassian.net/wiki/spaces/PROD/pages/4138500311/Templates+-+Segment+tracking

**_Learnings_**
* Typewriter and Segment are reliant on each other, but diverge a lot in types and package versions
* Typewriter's config doesn't really accommodate for different workspaces in Segment. 

**_Testing steps_**
* Follow the instructions to set up the additional configurations in the `analytics/readme` file
* Please check out the branch, run `yarn install` and confirm if the process completes successfully.
* Confirm the `trackingPlans` property has a valid ID in the generates `typewriter.yml` file
* Confirm `yarn typewriter:update` runs successfully
* (_optional_) Attempt to send a typewriter event to Segment.
